xivo_bus/base.py

```diff
@@ -1,29 +1,97 @@
 # Copyright 2021-2023 The Wazo Authors  (see the AUTHORS file)
 # SPDX-License-Identifier: GPL-3.0-or-later
 
+from __future__ import annotations
+
 import logging
-from collections import namedtuple
+from types import TracebackType
+from typing import Any, NamedTuple, Protocol
 
 from kombu import Exchange
 from kombu.utils.url import as_url
+from typing_extensions import Self
```

Comment:
`Self` type only availble in typing extensions from version 4.0.0 onward(https://github.com/python/typing_extensions/commit/0d10fec91740924bb48cc4435df093c12a1c2368).
bullseye packaging is version 3.7.4.3(https://packages.debian.org/bullseye/python3-typing-extensions).
Can this be a problem?

Reply:
I've removed it in a further commit, but will reintroduce it because we do backport bookworm's typing_extensions 4.4.0 in our mirrors

Reply:
Really nice to know. That means we can use typing_extensions to support generic TypedDict and NamedTuple(https://github.com/python/typing_extensions/blob/main/CHANGELOG.md#release-430-july-1-2022).